oak_dice/src/evidence.rs

```diff
@@ -0,0 +1,140 @@
+    /// signature of the next layer's certificate.
+    ///
+    /// The SHA2-256 digest of this field must included as additional data in the signed remote
+    /// attestation report to bind the key to the report. This means that the attestation
+    /// report effectively acts as a non-standard certificate for this key.
+    pub eca_public_key: [u8; PUBLIC_KEY_SIZE],
+}
+
+impl RootLayerEvidence {
+    pub fn get_tee_platform(&self) -> Option<TeePlatform> {
+        TeePlatform::from_repr(self.tee_platform)
+    }
+}
+
+static_assertions::assert_eq_size!([u8; 2320], RootLayerEvidence);
+
+/// Attestation evidence for non-root layers.
+#[derive(AsBytes, FromBytes)]
+#[repr(C)]
+pub struct LayerEvidence {
+    /// Serialized CWT certificate for the ECA private key owned by the corresponding layer. The
+    /// certificate must include measurements of the layer that owns the private key.
+    pub eca_certificate: [u8; CERTIFICATE_SIZE],
+}
+
+static_assertions::assert_eq_size!([u8; CERTIFICATE_SIZE], LayerEvidence);
+
+/// Private key that can be used by a layer to sign a certificate for the next layer.
+#[derive(AsBytes, FromBytes)]
+#[repr(C)]
+pub struct CertificateAuthority {
+    /// The RAW bytes representing an ECDSA private key.
+    pub eca_private_key: [u8; PRIVATE_KEY_SIZE],
+}
+
+static_assertions::assert_eq_size!([u8; PRIVATE_KEY_SIZE], CertificateAuthority);
+
+/// Wrapper for passing DICE info from Stage0 to the next layer (Stage 1 or the Restricted Kernel).
+#[derive(AsBytes, FromBytes)]
+#[repr(C, align(4096))]
+pub struct Stage0DiceData {
+    /// The evidence about Stage 0.
+    pub root_layer_evidence: RootLayerEvidence,
+    /// The evidence about the next layer.
+    pub layer_1_evidence: LayerEvidence,
+    pub layer_1_certifcate_authority: CertificateAuthority,
+    _padding: [u8; 688],
+}
+
+static_assertions::assert_eq_size!([u8; 4096], Stage0DiceData);
+
+/// ECDSA keys that can be used for an application for signing or encryption, and their associated
+/// certificates.
+#[derive(AsBytes, FromBytes)]
+#[repr(C)]
+pub struct ApplicationKeys {
+    /// The RAW bytes representing an ECDSA private key that can be used to sign arbitrary data.
+    pub signing_private_key: [u8; PRIVATE_KEY_SIZE],
+    /// Serialized CWT certificate for the signing private key. The certificate must include
+    /// measurements of the application.
+    pub signing_certificate: [u8; CERTIFICATE_SIZE],
+    /// The RAW bytes representing an ECDSA private key that can be used for hybrid encryption.
+    pub encryption_private_key: [u8; PRIVATE_KEY_SIZE],
+    /// Serialized CWT certificate for the encryption private key. The certificate must include
+    /// measurements of the application.
+    pub encryption_certificate: [u8; CERTIFICATE_SIZE],
+}
+
+static_assertions::assert_eq_size!([u8; 2176], ApplicationKeys);
+
+/// Wrapper for passing DICE info from the Restricted Kernel to the application.
```

Comment:
If it's passed from RKernel to the app, it may also contain the evidence for the application (i.e. `layer_2_evidence` or `application_evidence`).

And if we use `application_evidence` for this, we can also swap `layer_1_evidence` for `restricted_kernel_evidence`, since it's already a custom DICE data.

Reply:
The application evidence is already contained in the application_keys. Will rename layer_1_evidence to restricted_kernel_evidence.